src/commands/api-mesh/log-get-bulk.js

```diff
@@ -0,0 +1,208 @@
+		const timeDifferenceInSeconds = (endTime.getTime() - startTime.getTime()) / 1000;
+
+		if (timeDifferenceInSeconds > 1800) {
+			const hours = Math.floor(timeDifferenceInSeconds / 3600); //hours calculation
+			const minutes = Math.floor((timeDifferenceInSeconds % 3600) / 60); //minutes calculation
+			const seconds = timeDifferenceInSeconds % 60; //seconds calculation
+
+			this.error(
+				`Max duration between startTime and endTime should be 30 minutes. Current duration is ${hours} hour${
+					hours !== 1 ? 's' : ''
+				} ${minutes} minute${minutes !== 1 ? 's' : ''} and ${seconds} second${
+					seconds !== 1 ? 's' : ''
+				}.`,
+			);
+			return;
+		}
+
+		const { imsOrgId, imsOrgCode, projectId, workspaceId, workspaceName } = await initSdk({
+			ignoreCache,
+		});
+
+		// Retrieve meshId
+		let meshId = null;
+		try {
+			meshId = await getMeshId(imsOrgId, projectId, workspaceId, workspaceName);
+		} catch (err) {
+			this.error(`Unable to get mesh ID: ${err.message}`);
+		}
+
+		if (!meshId) {
+			this.error('Mesh ID not found.');
+		}
+
+		// 5. Call downloadFiles
+		const { presignedUrls, totalSize } = await downloadFilesSequentially(
+			imsOrgCode,
+			projectId,
+			workspaceId,
+			meshId,
+			formattedStartTime,
+			formattedEndTime,
+		);
+		//If presigned URLs are not found, throw error saying that no logs are found
+		if (!presignedUrls || presignedUrls.length === 0) {
+			this.error('No logs found for the given time range.');
+		}
+
+		let shouldDownload = false;
+		if (totalSize > 0) {
+			const totalSizeKB = (totalSize / 1024).toFixed(2); // Convert bytes to KB
+			// 7. Get user confirmation
+			shouldDownload = await promptConfirm(
+				`Expected file size is ${totalSizeKB} KB. Please confirm to output to file ${filename} (y/n)`,
+			);
+			if (shouldDownload) {
+				//create a writer and proceed with download
+				const writer = fs.createWriteStream(outputFile, { flags: 'a' });
+
+				// Write the column headers before appending the log content
+				writer.write(`${columnHeaders}\n`);
+
+				// Stream the data from the signed URLs
+				for (const urlObj of presignedUrls) {
+					const { key, url } = urlObj;
+					logger.info(`Downloading ${key} and appending to ${outputFile}...`);
+
+					try {
+						const fileContentStream = await this.downloadFileContent(url);
+						fileContentStream.pipe(writer, { end: false });
+
+						await new Promise((resolve, reject) => {
+							fileContentStream.on('end', resolve);
+							fileContentStream.on('error', reject);
+						});
+						// write a newline after each file write
+						writer.write('\n');
+
+						logger.info(`${key} content appended successfully.`);
+					} catch (error) {
+						logger.error(`Error downloading or appending content of ${key}:`, error);
+					}
+				}
+
+				this.log(`Successfully downloaded the logs to ${filename}`);
+			} else {
+				this.log('Log files are not downloaded.');
```

Comment:
This needs to be reworded, but I'm not certain of the context. i.e. Why were the files not downloaded?

Reply:
@jhadobe, the context is that we download and write to the file only after receiving confirmation from the user based on the expected log file size. So if they answer `no`, we just return this message